azure/hub/main.tf

```
@@ -0,0 +1,102 @@
+#################
+# Resource Group
+#################
+
+resource "azurerm_resource_group" "main" {
+  name     = var.resource_group_name
+  location = var.location
+  tags     = var.tags
+}
+
+##########
+# Hub VNet
+##########
+
+module "network" {
+  source  = "ans-coe/virtual-network/azurerm"
+  version = "1.3.0"
+
+  name                = var.virtual_network_name
+  location            = var.location
+  resource_group_name = azurerm_resource_group.main.name
+  tags                = var.tags
+
+  address_space = var.address_space
+  subnets = merge(
+    local.enable_private_endpoint_subnet ? {
+      (var.private_endpoint_subnet["subnet_name"]) = {
+        prefix                                        = var.private_endpoint_subnet["subnet_prefix"]
+        service_endpoints                             = var.private_endpoint_subnet["service_endpoints"]
+        private_endpoint_network_policies_enabled     = false
+        private_link_service_network_policies_enabled = false
+      }
+    } : {},
+
```

Noticing firewall is not present here as it's handled by the firewall module. While I'd suggest that the firewall module should consume a subnet_id var instead of creating its own subnet, I would at least suggest placing a comment here to indicate that we're handling the firewall subnet in this fashion.

```suggestion
    # The firewall subnet is managed by the module found in firewall.tf
```